pkg/resources/managed_account.go

```diff
@@ -102,82 +96,109 @@ func ManagedAccount() *schema.Resource {
 
 // CreateManagedAccount implements schema.CreateFunc.
 func CreateManagedAccount(d *schema.ResourceData, meta interface{}) error {
-	return CreateResource(
-		"this does not seem to be used",
-		managedAccountProperties,
-		managedAccountSchema,
-		snowflake.NewManagedAccountBuilder,
-		initialReadManagedAccount,
-	)(d, meta)
-}
+	db := meta.(*sql.DB)
+	ctx := context.Background()
+	client := sdk.NewClientFromDB(db)
+
+	name := d.Get("name").(string)
+	id := sdk.NewAccountObjectIdentifier(name)
+
+	adminName := d.Get("admin_name").(string)
+	adminPassword := d.Get("admin_password").(string)
+	createParams := sdk.NewCreateManagedAccountParamsRequest(adminName, adminPassword)
+
+	if v, ok := d.GetOk("comment"); ok {
+		createParams.WithComment(sdk.String(v.(string)))
+	}
+
+	createRequest := sdk.NewCreateManagedAccountRequest(id, *createParams)
+
+	err := client.ManagedAccounts.Create(ctx, createRequest)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(helpers.EncodeSnowflakeID(id))
 
-// initialReadManagedAccount is used for the first read, since the locator takes
-// some time to appear. This is currently implemented as a sleep. @TODO actually
-// wait until the locator is generated.
-func initialReadManagedAccount(d *schema.ResourceData, meta interface{}) error {
-	log.Println("[INFO] sleeping to give the locator a chance to be generated")
-	// lintignore:R018
-	time.Sleep(10 * time.Second)
 	return ReadManagedAccount(d, meta)
 }
 
 // ReadManagedAccount implements schema.ReadFunc.
 func ReadManagedAccount(d *schema.ResourceData, meta interface{}) error {
 	db := meta.(*sql.DB)
-	id := d.Id()
-
-	stmt := snowflake.NewManagedAccountBuilder(id).Show()
-	row := snowflake.QueryRow(db, stmt)
-	a, err := snowflake.ScanManagedAccount(row)
-
-	if errors.Is(err, sql.ErrNoRows) {
-		// If not found, remove resource from
-		log.Printf("[DEBUG] managed account (%s) not found", d.Id())
-		d.SetId("")
-		return nil
-	}
+	client := sdk.NewClientFromDB(db)
+
+	ctx := context.Background()
+	objectIdentifier := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
+
+	// We have to wait during the first read, since the locator takes some time to appear.
+	// This approach has a downside of not handling correctly the situation where managed account was removed externally.
+	// TODO [SNOW-1003380]: discuss it as a provider-wide topic during resources redesign.
+	var managedAccount *sdk.ManagedAccount
+	var err error
```

Comment:
why err is predefined ?

Reply:
because managedAccount has to be predefined and then you either:
- predefined err here
- predefined err inside retry (because you can't use `:=` syntax with predefined and not predefined var)

either way you predefine it somewhere